### PR TITLE
Systemverilog: adapt to current Surelog

### PR DESCRIPTION
**Pull request by rkapuscik**
* Update `import_typespec` case
* Handle shortint and time typespecs
* Fix packed ranges access in nets

### CODE REVIEW

File: systemverilog-plugin/UhdmAst.cc [hunk 1]
```diff
@@ -3352,6 +3357,32 @@ void UhdmAst::process_int_typespec()
...
 
+void UhdmAst::process_shortint_typespec()
+{
+    std::vector<AST::AstNode *> packed_ranges;   // comes before wire name
+    std::vector<AST::AstNode *> unpacked_ranges; // comes after wire name
+    current_node = make_ast_node(AST::AST_WIRE);
+    auto left_const = AST::AstNode::mkconst_int(16, true);
+    auto right_const = AST::AstNode::mkconst_int(0, true);
+    auto range = new AST::AstNode(AST::AST_RANGE, left_const, right_const);
+    packed_ranges.push_back(range);
```
Comment (kamilrakoczy) [inline systemverilog-plugin/UhdmAst.cc:3368]: Please replace with ``make_range(16, 0)``: https://github.com/chipsalliance/yosys-f4pga-plugins/blob/main/systemverilog-plugin/UhdmAst.cc#L96

File: systemverilog-plugin/UhdmAst.cc [hunk 2]
```diff
@@ -3352,6 +3357,32 @@ void UhdmAst::process_int_typespec()
...
+}
+
+void UhdmAst::process_time_typespec()
+{
+    std::vector<AST::AstNode *> packed_ranges;   // comes before wire name
+    std::vector<AST::AstNode *> unpacked_ranges; // comes after wire name
+    current_node = make_ast_node(AST::AST_WIRE);
+    auto left_const = AST::AstNode::mkconst_int(64, true);
+    auto right_const = AST::AstNode::mkconst_int(0, true);
+    auto range = new AST::AstNode(AST::AST_RANGE, left_const, right_const);
```
Comment (kamilrakoczy) [inline systemverilog-plugin/UhdmAst.cc:3380]: Please replace with ``make_range``.

**Review by kamilrakoczy — CHANGES_REQUESTED** [review]

File: systemverilog-plugin/UhdmAst.cc (hunk 1 above)
Comment (rkapuscik) [reply systemverilog-plugin/UhdmAst.cc]: Replaced, also changed this in another typespec handler.

File: systemverilog-plugin/UhdmAst.cc (hunk 2 above)
Comment (rkapuscik) [reply systemverilog-plugin/UhdmAst.cc]: Done

**Review by kamilrakoczy — APPROVED** [review]

**Comment by hzeller** [comment]
Is this ready to merge ?

**Comment by hzeller** [comment]
Are there bugs filed in Surelog for the parameter simplification issues ?

**Pull request merged**